src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -53,6 +53,18 @@ class PipelineInfo:
     failures: str
 
 
+@dataclass
+class AzureServicePrincipalInfo:
+    active: bool
+    application_id: str
+    display_name: str
+    entitlements: list
+    external_id: str
+    groups: list
+    spn_id: str
+    roles: list
```

Comment:
We don't support list columns in our inventory tables yet

Reply:
added logic in def _assess_pipelines(self, all_pipelines):. Can you please review. If ok, then I will add the same in a module for other place - such as cluster ,job cluster, etc.